### PR TITLE
Switch to explicit semver type

### PR DESCRIPTION
**Pull request by vosst**

### CODE REVIEW

File: cmds/db-manager/main.go [hunk 1]
```diff
@@ -239,8 +246,6 @@ func (m *MyMigrate) MigrationDirection(desiredVersion string, desiredStep int) (
 	if err != nil {
 		return 0, fmt.Errorf("Failed to get current DB version to determine migration direction: %v", err)
 	}
-	if !semver.IsValid(currentVersion) {
-		return 0, fmt.Errorf("The current DB Version format is in valid")
-	}
-	return Direction(semver.Compare(desiredVersion, currentVersion)), nil
+
+	return Direction(desiredVersion.Compare(*currentVersion)), nil
```
Comment (Dos-Ph) [inline cmds/db-manager/main.go:250]: Do you have to cover the case where desiredVersion == nil?

**Comment by Dos-Ph** [comment]
> We will need to also change the db-schemas/*.sql files where we bump the version numbers as well

I don't think we need. We just make sure that the new version after this doesn't contain 'v'.

File: cmds/db-manager/main.go (hunk 1 above)
Comment (vosst) [reply cmds/db-manager/main.go:250]: Nope, `desiredVersion` is not a pointer.

**Comment by vosst** [comment]
> We will need to also change the db-schemas/*.sql files where we bump the version numbers as well

The code handles versions with and without prefix `v`.

**Comment by vosst** [comment]
@charlie-pisuraj ping 😄

**Pull request merged**